### PR TITLE
chore(context): add label as API query param

### LINKED ISSUE #403: are my usecases valid for this workflow?
I get this error [HttpError: API rate limit exceeded for installation ID 9256508](https://github.com/FreeTubeApp/FreeTube/runs/5961793366?check_suite_focus=true) on all our workflows after i added this workflow to our [repo](https://github.com/FreeTubeApp/FreeTube/actions/workflows/dependent-issues.yml?query=is%3Afailure). It tries to check every open issue and PR for dependencies and i assume that is the reason why we are exceeding the api limit. 

i had these usecases in mind when i merged over the workflow in our repo. 

1. PR1 is dependent on PR2 (PR2 has to be merged/closed first in order to get PR1 merged). This one is pretty straight forward.

2. PR1 is dependent on issue1 (issue1 has to be closed first in order to get PR1 merged). This can be helpful when someone already created PR1 but u want issue1 to get addressed first but nobody has created a PR for it yet.

3. issue1 is dependent on PR1 (PR1 has to be merged/closed first in order to close issue1). This can be helpful when somebody forgets to link PR1 as dependency when they create PR2 for issue1

4. issue1 is dependent on issue2 (issue2 has to be closed first in order to close issue1). This can be helpful when there are no PR's yet that addresses one of the issues.

Are the usecases above valid for this workflow? if so is there a way to not hit the api limit so fast?

### PR DESCRIPTION
**Pull request by johncardiologs**
# Description

When the action is running outside of the context of an issue/PR (typically when we close an issue/PR), we are currently checking all open issues/PRs to see if any of them depend on the issue/PR that is being closed. This can lead to hitting the github API rate limit of 500 requests in one go (there are many repos that have 500+ combined issues/PRs out there).

For starters, we can check against all open PRs that have the `depends` label associated with them, as the user has defined this label in the action.yaml file's `label` field.

This is a simplistic fix but should make it less likely to hit the github API rate limit of 500 requests (hopefully the amount of open PRs + issues that have the `depends` label in the user's repo is less than 500).

A better fix would be that the issue/PR that is closed knows exactly what other issues/PRs depend on it, which I don't believe is currently implemented, but it took me a while to figure out the github API and that's all I have in me for today.

# Motivation

Closes #403

### CODE REVIEW

File: src/context.ts [hunk 1]
```diff
@@ -70,12 +72,16 @@ export async function getActionContext(): Promise<ActionContext> {
 			per_page: 100,
 		};
 
+		let queryParams = config.label
+			? { ...options, labels: config.label }
+			: options;
+
```
Comment (johncardiologs) [inline src/context.ts:78]: I realized the label provided in the config is optional.

If it is not provided, `config.label` (as returned by `core.getInput`) will be an empty string, so the API query will default to the existing `options` variable. If `config.label` is provided, we set the `labels` field on the query.

**Comment by johncardiologs** [comment]
Hi @z0al , any updates on this? Happy to clarify further if there are any questions.

If you set this up on a local repo, when closing a dependent issue / PR, you'll notice in the CI logs that only issues that are open & that have the dependent label will be checked. Any other open issues/PRs that don't have the label will not be checked.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@z0al

**Review by z0al — APPROVED** [review]

**Comment by z0al** [comment]
Sorry for the delay everyone. I don't this action anymore so I barely ever check this repo. For everyone's sake, I'm going to archive the repo. 

Of course, anyone could always fork and maintain their own thing.

Best

**Pull request merged**